webui/react/src/hooks/useSettingsProvider.tsx

```diff
@@ -0,0 +1,105 @@
+import { Map } from 'immutable';
```

Comment:
question: what's the advantage of using Map from `immutable`?
The map just becomes immutable? Whats the difference between them in terms of behavior?

Reply:
It's more performant for immutable usecases (which this is). Mutably updating a map in react state introduces potential racing issues.

Reply:
How about using [`Immer`](https://www.npmjs.com/package/immer) instead of `immutable` then? I think its pretty much the same for our usage, and its smaller bundle and more active on github

Reply:
I don't know much about immer but immutable.js has less github activity because it's stable. If you'd like to evaluate different datatype libraries add a ticket and maybe we can swap them out. For the purpose of this PR immutable works and it's been the de-facto standard for a while.